inference-engine/src/transformations/src/transformations/common_optimizations/clamp_fusion.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/clamp_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ClampFusion, "ClampFusion", 0);
+
+ngraph::pass::ClampFusion::ClampFusion() {
+    auto data_pattern = ngraph::pattern::any_input();
+    auto min_const_pattern = ngraph::pattern::wrap_type<opset5::Constant>();
+    auto max_const_pattern = ngraph::pattern::wrap_type<opset5::Constant>();
+    auto max_pattern = ngraph::pattern::wrap_type<opset5::Maximum>({data_pattern, min_const_pattern});
+    auto min_pattern = ngraph::pattern::wrap_type<opset5::Minimum>({max_pattern, max_const_pattern});
+
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto pattern_map = m.get_pattern_value_map();
+        auto data = pattern_map.at(data_pattern);
+        auto min_const = std::dynamic_pointer_cast<opset5::Constant>(pattern_map.at(min_const_pattern).get_node_shared_ptr());
+        if (!min_const)
+            return false;
+        if (shape_size(min_const->get_shape()) != 1)
```

Comment:
It returns with false for scalar case. You can check `is_scalar(min_const->get_shape())` (or change condition to `shape_size(min_const->get_shape()) >= 1`)

Reply:
no it doesn't. shape_size for scalars is 1 (https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/ngraph/shape.hpp#L71). I could do is_scalar(...), but I wanted to allow both Shape{1} and Shape{} to match this pattern.